common/src/main/java/org/apache/falcon/entity/parser/ClusterEntityParser.java

```diff
@@ -243,37 +244,72 @@ private void validateLocations(Cluster cluster) throws ValidationException {
         try {
             fs = HadoopClientFactory.get().createProxiedFileSystem(conf);
         } catch (FalconException e) {
-            throw new ValidationException(
-                    "Unable to get file system handle for cluster " + cluster.getName(), e);
+            throw new ValidationException("Unable to get file system handle for cluster " + cluster.getName(), e);
         }
 
-        for (Location location : cluster.getLocations().getLocations()) {
-            final String locationName = location.getName();
-            if (locationName.equals("temp")) {
-                continue;
-            }
+        if (ClusterHelper.getLocation(cluster, ClusterLocationType.STAGING) == null) {
+            throw new ValidationException(
+                    "Unable to find the mandatory location of name: " + ClusterLocationType.STAGING.value()
+                            + " for cluster " + cluster.getName());
+        } else {
 
+            Location stagingLocation = ClusterHelper.getLocation(cluster, ClusterLocationType.STAGING);
```

Comment:
You can call ClusterHelper.getLocation(cluster, ClusterLocationType.STAGING) once and re-use later